tests/test_views.py

```diff
@@ -0,0 +1,1050 @@
+        project = factories.ProjectFactory()
+        url = factories.URLFactory(
+            url_project=project,
+            attribute='surt',
+            value=surt
+        )
+        response = client.get(reverse('url_surt', args=[project.project_slug, surt_root]))
+
+        assert response.context['surt'] == surt_root
+        assert response.context['project'] == project
+        assert len(response.context['url_list']) == url_list_len
+        if url_list_len == 1:
+            assert response.context['url_list'][0] == url
+        assert response.context['letter'] == letter
+        assert response.context['browse_domain'] == 'com'
+        assert len(response.context['browse_dict']) == 1
+        for each in response.context['browse_dict']['com']:
+            if each[1] is not None:
+                assert each == ('E', 'http://(com,e')
+
+
+class TestUrlAdd():
+
+    @pytest.mark.parametrize('request_type, data_dict', [
+        ('get', {}),
+        ('post', {
+            'nominator_name': 'Eddie',
+            'nominator_institution': 'UNT',
+            'nominator_email': 'someone@somewhere.com'
+        })
+    ])
+    def test_status_ok(self, client, request_type, data_dict):
+        project = factories.ProjectFactory()
+        response = getattr(client, request_type)(
+            reverse('url_add', args=[project.project_slug]), data_dict)
+        assert response.status_code == 200
+
+    def test_template_used(self, client):
+        project = factories.ProjectFactory(registration_required=False)
+        response = client.post(
+            reverse('url_add', args=[project.project_slug]),
+            {
+                'nominator_name': 'Eddie',
+                'nominator_institution': 'UNT',
+                'nominator_email': 'someone@somewhere.com'
+            }
+        )
+
+        assert response.templates[0].name == 'nomination/url_add.html'
+
+    def test_base_context_values(self, client):
+        entity = u'http://www.example.com'
+        entity_surt = u'http://(com,example,www,)'
+        project = factories.ProjectFactory()
+        project_metadata = factories.ProjectMetadataFactory(project=project)
+        factories.URLFactory(url_project=project, entity=entity)
+        factories.SURTFactory(
+            url_project=project,
+            entity=entity,
+            value=entity_surt
+        )
+        factories.SURTFactory.create_batch(
+            1,
+            url_project=project,
+            entity=u'{}/main/page'.format(entity),
+            value=u'{}/main/page'.format(entity_surt)
+        )
+        expected_context = {
+            'project': project,
+            'form': views.URLForm(),
+            'metadata_vals': views.get_metadata(project),
+            'institutions': views.get_look_ahead(project),
+            'form_types': json.dumps({project_metadata.metadata.name: project_metadata.form_type}),
+        }
```

Comment:
It looks like `metadata_vals`, `form_types`, `institutions`, and `url_data` are all duplicated